### PR TITLE
CLN: replacing '.format' with f-strings in various files

### PR DESCRIPTION
**Pull request by AlfredoGJ**
- [x]  contributes to #29547
- [ ] tests added / passed
- [X] passes `black pandas`
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: scripts/validate_docstrings.py [hunk 1]
```diff
@@ -938,13 +938,9 @@ def header(title, width=80, char="#"):
         full_line = char * width
         side_len = (width - len(title) - 2) // 2
         adj = "" if len(title) % 2 == 0 else " "
-        title_line = "{side} {title}{adj} {side}".format(
-            side=char * side_len, title=title, adj=adj
-        )
+        title_line = f"{char * side_len} {title}{adj} {side}"
```
Comment (MomIsBestFriend) [inline scripts/validate_docstrings.py]: Replace the ```{side}``` with ```{char * side_len}``` :smile:

File: pandas/tseries/holiday.py [hunk 2]
```diff
@@ -394,8 +394,8 @@ def holidays(self, start=None, end=None, return_name=False):
         """
         if self.rules is None:
             raise Exception(
-                "Holiday Calendar {name} does not have any "
-                "rules specified".format(name=self.name)
+                f"Holiday Calendar {self.name} does not have any "
+                f"rules specified"
```
Comment (simonjayhawkins) [inline pandas/tseries/holiday.py]: f string prefix not needed here
```suggestion
                "rules specified"
```

File: scripts/generate_pip_deps_from_conda.py [hunk 3]
```diff
@@ -127,13 +127,13 @@ def main(conda_fname, pip_fname, compare=False):
     )
     if res:
         msg = (
-            "`requirements-dev.txt` has to be generated with `{}` after "
-            "`environment.yml` is modified.\n".format(sys.argv[0])
+            f"`requirements-dev.txt` has to be generated with `{sys.argv[0]}` after "
+            f"`environment.yml` is modified.\n"
```
Comment (simonjayhawkins) [inline scripts/generate_pip_deps_from_conda.py]: same

File: scripts/generate_pip_deps_from_conda.py [hunk 4]
```diff
@@ -127,13 +127,13 @@ def main(conda_fname, pip_fname, compare=False):
...
         )
         if args.azure:
             msg = (
-                "##vso[task.logissue type=error;"
-                "sourcepath=requirements-dev.txt]{}".format(msg)
+                f"##vso[task.logissue type=error;"
```
Comment (simonjayhawkins) [inline scripts/generate_pip_deps_from_conda.py]: same

**Review by simonjayhawkins — COMMENTED** [review]
@AlfredoGJ Thanks for the PR. some minor nits otherwise lgtm.

File: scripts/validate_docstrings.py (hunk 1 above)
Comment (AlfredoGJ) [reply scripts/validate_docstrings.py]: Now I see, thank you.

File: scripts/generate_pip_deps_from_conda.py (hunk 3 above)
Comment (AlfredoGJ) [reply scripts/generate_pip_deps_from_conda.py]: Thanks for your feedback, I just changed it

File: pandas/tseries/holiday.py [hunk 5]
```diff
@@ -394,8 +394,7 @@ def holidays(self, start=None, end=None, return_name=False):
         """
         if self.rules is None:
             raise Exception(
-                "Holiday Calendar {name} does not have any "
-                "rules specified".format(name=self.name)
+                f"Holiday Calendar {self.name} does not have any " "rules specified"
```
Comment (datapythonista) [inline pandas/tseries/holiday.py]: You'll have to join these two strings

**Review by datapythonista — COMMENTED** [review]
Thanks @AlfredoGJ, just one small thing, and this can be merged.

**Review by simonjayhawkins — APPROVED** [review]
@AlfredoGJ lgtm pending green.

File: pandas/tseries/holiday.py (hunk 5 above)
Comment (AlfredoGJ) [reply pandas/tseries/holiday.py]: Thanks for the feedback

**Comment by jreback** [comment]
thanks @AlfredoGJ

**Pull request merged**